randovania/gui/dialog/logic_settings_window.py

```diff
@@ -432,6 +460,20 @@ def _on_check_starting_area(self, check, _):
                                                                                   check.isChecked())
             )
 
+    def _on_check_starting_world(self, check, _):
+        if self._during_batch_check_update:
+            return
+        world_list = self.game_description.world_list
+        world = world_list.world_by_asset_id(check.world_asset_id)
+        world_areas = [world_list.area_to_area_location(area)
+                       for area in world.areas if check.is_dark_world == area.in_dark_aether]
+        with self._editor as editor:
+            for area in world_areas:
+                editor.set_layout_configuration_field(
```

Comment:
`set_layout_configuration_field` does extra boiler plate which is a waste inside a for loop like this